File: apps/telegram_mvp_kmp/shared/src/commonMain/kotlin/com/hypheng/telegrammvpkmp/ui/BootstrapScreen.kt
package com.hypheng.telegrammvpkmp.ui

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.size
import androidx.compose.material3.CircularProgressIndicator
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp

import com.hypheng.telegrammvpkmp.design.DesignAssetPaths

@Composable
fun BootstrapScreen(brandTitle: String = "Telegram Demo") {
    Column(
        modifier = Modifier.fillMaxSize(),
        horizontalAlignment = Alignment.CenterHorizontally,
        verticalArrangement = Arrangement.Center,
    ) {
        DesignSvgIcon(
            assetPath = DesignAssetPaths.telegramBrandBadge,
            modifier = Modifier.size(80.dp),
            size = 80.dp,
        )
        Text(
            text = brandTitle,
            style = MaterialTheme.typography.headlineMedium,
        )
        CircularProgressIndicator()
        Text(text = "Checking local session…")
    }
}

File: apps/telegram_mvp_flutter/lib/src/features/bootstrap/bootstrap_screen.dart
import 'package:flutter/material.dart';

import '../../design/design_assets.dart';
import '../../widgets/design_svg_icon.dart';
import '../../app/app_controller.dart';

class BootstrapScreen extends StatelessWidget {
  const BootstrapScreen({super.key, required this.controller});

  final TelegramAppController controller;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: Column(
          mainAxisSize: MainAxisSize.min,
          children: [
            DesignSvgIcon(DesignAssetPaths.telegramBrandBadge, size: 80),
            const SizedBox(height: 20),
            Text(
              controller.catalog?.login.brandTitle ?? 'Telegram Demo',
              style: const TextStyle(fontSize: 28, fontWeight: FontWeight.w800),
            ),
            const SizedBox(height: 12),
            const SizedBox(
              width: 26,
              height: 26,
              child: CircularProgressIndicator(strokeWidth: 2.6),
            ),
            const SizedBox(height: 16),
            Text(
              'Checking local session…',
              style: TextStyle(
                fontSize: 14,
                color: Theme.of(context).colorScheme.onSurfaceVariant,
              ),
            ),
          ],
        ),
      ),
    );
  }
}
